packages/node_modules/@node-red/editor-client/src/js/ui/common/typedInput.js

```diff
@@ -101,6 +108,111 @@
             return matches;
         }
     }
+    
+    let contextKnownKeys = {}
+    let contextCache = {}
+    if (RED.events) {
+        RED.events.on("editor:close", function () {
+            contextCache = {}
+            contextKnownKeys = {}
+        });
+    }
+
+
+    const contextAutoComplete = function(options) {
+
+        const getContextKeysFromRuntime = function(scope, store, searchKey, done) {
+            contextKnownKeys[store] = contextKnownKeys[store] || new Set()
```

Comment:
I would add `scope` to be able to play with `flow` and `global` inputs

```suggestion
            contextKnownKeys[scope] = contextKnownKeys[scope] || {}
            contextKnownKeys[scope][store] = contextKnownKeys[scope][store] || new Set()
```

Reply:
Ah yes - good spot. Hadn't thought of that when I made the cache shared.

Reply:
I only added it to the definition - don't forget to change it everywhere